compiler/psi/src/org/jetbrains/kotlin/parsing/KotlinExpressionParsing.java

```diff
@@ -1493,6 +1493,11 @@ private void parseFor() {
                 error("Expecting a variable name");
             }
 
+            TokenSet recoverySet = TokenSet.create(RPAR, LBRACE, RBRACE, EOL_OR_SEMICOLON, EOF);
+            if (!atSet(recoverySet)) {
+                errorUntil("Unexpected tokens in loop", recoverySet);
```

Comment:
Minor: I believe `in "for" loop` would be clearer than `in loop`.

Reply:
`in "the" for loop`